### ISSUE TITLE
Move service's status to manager

### ISSUE THREAD

**Issue by NicolasMahe** [mesg-foundation/engine#958]
In PR https://github.com/mesg-foundation/core/pull/904, the package manager has been introduced to manage the logic between service and container.

Continuation of conversation: https://github.com/mesg-foundation/core/pull/904#discussion_r277137119

Now that's the manager is managing all container related functions, the status of a service should not stay in the package service but should move to the package manager.

The service package should have only logic about the service's data and nothing more.

Basically, the file [`manager.go`](https://github.com/mesg-foundation/core/pull/904/files#diff-0cae5fb20442a4fd1b527406e43c6d8a) should only return struct define in its own package.

Please check the conversation first, and then tell if you are agree to move namespace to manager or leave it like this.

**Comment by ilgooz** [comment]
I can't quite agree with this one because `manager.go` is also depends on the `service.Service`, in this case we have to move it to manager as well. In that case, there is no reason for having both `service` & `manager` packages. because they should be merged into one. to me, `service` pkg is all about storing the data types for services and other useful stuff like validation, getters, errors, etc.

**Comment by NicolasMahe** [comment]
`service.Service` is passed as inputs of the functions in manager.
I'm speaking about the returned values of the manager's function.

> to me, `service` pkg is all about storing the data types for service and useful stuff for service like validation, getters, errors types etc.

So why service should keep status when its logic is implemented in another package?

**Comment by ilgooz** [comment]
it's a logical decision. Manager is responsible with dealing with `Service` type and its siblings, not defining them. because Manager has nothing to do with service validation, task getter, event not found errors or etc. its just responsible for managing the service's container and providing logs, statuses, etc.

**Comment by NicolasMahe** [comment]
I see the manager package as the link between service and container.
Service should not have anything about container.
Container should not know anything about service.
Only manager knows both and define the logic and data structure that "connect" service with container.

What is the status of a service if service doesn't know anything about container?
Same for log.

Maybe manager is not the best name for the manager package as it's too generic and doesn't describe enough the goal of the package.

@antho1404 suggested "runner" in the PR https://github.com/mesg-foundation/core/pull/904. I kind of like it. Maybe with a bit more context: "serviceRunner".

@mesg-foundation/core  What do you think?

**Comment by antho1404** [comment]
Container and Service should no know each other and yes only the manager knows both.

For now, we get the status based on the container information so service cannot do that and the manager should do it because it knows both.

For the name, I don't really like the name Manager, even serviceRunner is not ideal but still better. Moving this package into api (that controls all the logic) and calling it service would make sense.

This way it's an api package that manages the service that runs and we can call it instances or serviceInstances to start the next developments.

This for me makes a lot of sense and totally remove the dependency between the service package and the container. Also a good argument for that is that all the business logic is already fully delegated to this manager in the api meaning that this business logic is really related to the api

**Comment by NicolasMahe** [comment]
> Moving this package into api (that controls all the logic) and calling it service would make sense.

Totally agree ;) but still let's keep it at a separate package like: `api/service`.

**Comment by krhubert** [comment]
> @antho1404 suggested "runner" in the PR #904. I kind of like it. Maybe with a bit more context: "serviceRunner".

👎 - runner sounds like js runner or task runner something like this. Manager tells the story. Docker container could be called runner because it **runs** the service but don't change the manager name (at least not for a runner).

The things you are talking about are in the https://github.com/mesg-foundation/core/pull/799/ ;) so yes and yes. The service shouldn't know anything about the container and vice versa and the `Manager` should be link between them
